src/shared/event-array.js

```diff
@@ -48,6 +50,12 @@ export default class EventArray {
     this._data[index] = value;
   }
 
+  unshift(...values) {
+    this._data.unshift(...values);
+    this.length += values.length;
+    this._length += values.length;
+  }
```

Comment:
we ended up not needing unshift, so I'd back this back out, as it turns out that this is problematic for the events array as we use index to track start/stop/resume